core/logger/logger.go

```diff
@@ -211,10 +247,81 @@ func WithFile(file io.Writer) Option {
 	}
 }
 
+// WithFileRotation sets up the option to roll log files
+func WithFileRotation(prefix string, subfolder string, fileExtension string) Option {
+	return func(el *Logger) error {
+		file, err := newFile(prefix, subfolder, fileExtension)
+		if err != nil {
+			return err
+		}
+
+		el.roll = true
+		el.file.prefix = prefix
+		el.logFiles = append(el.logFiles, file)
+		el.file.SetWriter(file)
+		return nil
+	}
+}
+
+// WithStderrRedirect sets up the option to redirect stderr to file
+func WithStderrRedirect() Option {
+	return func(el *Logger) error {
+		file, err := newFile("fatalErrors"+el.file.prefix, "logs", "log")
+		if err != nil {
+			return err
+		}
+
+		err = redirectStderr(file)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+}
+
 // WithStackTraceDepth sets up the stackTraceDepth option for the Logger
 func WithStackTraceDepth(depth int) Option {
 	return func(el *Logger) error {
 		el.stackTraceDepth = depth
 		return nil
 	}
 }
+
+func (el *Logger) rollFiles() {
+	duration := time.Now().Sub(el.file.creationTime).Seconds()
+	if duration > fileLifetimeInSeconds {
+		if len(el.logFiles) == nrOfFilesToRemember {
+			_ = el.logFiles[0].Close()
```

Comment:
This means we will get to 24 open files? We are only using the current one, maybe we can close the previous one each time we are creating a new file.

Reply:
Done.